File: message/tagsequencedmessage.go
package message

import "github.com/CotaPreco/Horus/tag"

type TagSequencedMessage struct {
	Message
	Tags []tag.Tag
}

func NewTagSequencedMessage(tags []tag.Tag, payload []byte) *TagSequencedMessage {
	return &TagSequencedMessage{
		Message{
			payload,
		},
		tags,
	}
}

File: php/src/CotaPreco/Horus/Message/TagSequencedMessage.php
<?php

namespace CotaPreco\Horus\Message;

use CotaPreco\Horus\Tag\TagInterface;

/**
 * @author Andrey K. Vital <andreykvital@gmail.com>
 */
class TagSequencedMessage extends Message
{
    /**
     * @var TagInterface[]
     */
    private $tags;

    /**
     * @param TagInterface[] $tags
     * @param string         $message
     */
    public function __construct(array $tags, $message)
    {
        $this->tags = array_map(
            function (TagInterface $tag) {
                return $tag;
            },
            $tags
        );

        parent::__construct($message);
    }

    /**
     * @return TagInterface[]
     */
    public function getTags()
    {
        return $this->tags;
    }
}
